### PR TITLE
feat: Add a rule that stations (location_type 1) must be the parent_station of some stop (location_type 0).

### PR DESCRIPTION
**Pull request by bradyhunsaker**
**Summary:**

Part of #153.

The new notice is of type INFO because there is nothing in the specification or best practices stating that stations should have stops. There is a discussion about changing this in #1348.

As part of the change, this renames ParentLocationTypeValidator to the more general name ParentStationValidator.

For unit testing it is now necessary to consider two types of notices, so all "contains()" calls are changed to "containsExactly()" or "containsExactlyElementsIn()" to be sure that additional notices are not ignored.

**Expected behavior:** 

If a location of type 1 (station) never appears as the parent station of a location of type 0 (stop/platform), then issue a notice.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [ X] Run the unit tests with `gradle test` to make sure you didn't break anything
- [ X] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [X ] Linked all relevant issues
- [ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)

### CODE REVIEW

**Review by davidgamez — CHANGES_REQUESTED** [review]
Hi @bradyhunsaker, few changes have been merged since your first PR regarding the notice documentation generation that impacts your contribution. I added a few comments to align this PR with the latest changes before approval.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ParentStationValidator.java [hunk 1]
```diff
@@ -156,4 +186,22 @@ static class WrongParentLocationTypeNotice extends ValidationNotice {
...
+
+  /**
+   * Unused parent station.
+   *
+   * <p>Severity: {@code SeverityLevel.INFO}
+   */
+  static class UnusedParentStationNotice extends ValidationNotice {
+    private final int csvRowNumber;
+    private final String stopId;
+    private final String stopName;
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ParentStationValidator.java:198]: As part of #1324, now the `ValidationNotice` classes contain their own documentation. We will be deprecating the rules.md soon in favor of inline documentation and GtfsValidationNotice annotation. More information regarding the `new pattern` for adding a notice can be found at [NEW_RULES.md](https://github.com/MobilityData/gtfs-validator/blob/master/docs/NEW_RULES.md)
Find below a suggested change to align this PR with the latest documentation requirements:
```suggestion
  /**
   * Unused parent station.
   *
   * <p>A stop has `location_type` STATION (1) but does not appear in any stop's `parent_station`.
   */
  @GtfsValidationNotice(
          severity = SeverityLevel.INFO,
          files = @FileRefs({GtfsStopSchema.class}))
  static class UnusedParentStationNotice extends ValidationNotice {
    /** The row number of the faulty record. */
    private final int csvRowNumber;

    /** The id of the faulty stop. */
    private final String stopId;

    /** The name of the faulty stop. */
    private final String stopName;
```

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ParentStationValidator.java [hunk 2]
```diff
@@ -156,4 +186,22 @@ static class WrongParentLocationTypeNotice extends ValidationNotice {
...
+
+    UnusedParentStationNotice(int csvRowNumber, String stopId, String stopName) {
+      super(SeverityLevel.INFO);
```
[the remark this answers is not in the document]
Comment (bradyhunsaker) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ParentStationValidator.java]: Done.

**Comment by bradyhunsaker** [comment]
Thanks, David! I accepted your suggested edit and removed the super() call. Any other comments?

**Comment by davidgamez** [comment]
> Thanks, David! I accepted your suggested edit and removed the super() call. Any other comments?

Thanks, @bradyhunsaker. One more ask, Can you review the formatting(_./gradlew goJF_) and class imports?  I'm sorry I typed my suggestions on Github and didn't have the proper formatting and missed an import.

**Comment by bradyhunsaker** [comment]
OK. I'm wrestling with google-java-format right. I'm getting an error like this one:
https://github.com/sherter/google-java-format-gradle-plugin/issues/68

I'll try again tomorrow.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ParentStationValidator.java [hunk 3]
```diff
@@ -156,4 +186,29 @@ static class WrongParentLocationTypeNotice extends ValidationNotice {
...
   }
+
+  /**
+   * Unused parent station.
+   *
+   * <p>A stop has `location_type` STATION (1) but does not appear in any stop's `parent_station`.
+   */
+  @GtfsValidationNotice(
+          severity = SeverityLevel.INFO,
+          files = @FileRefs({GtfsStopSchema.class}))
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ParentStationValidator.java:197]: @bradyhunsaker you can try this:
```suggestion
  @GtfsValidationNotice(severity = SeverityLevel.INFO, files = @FileRefs({GtfsStopSchema.class}))
```
* also make sure SeverityLevel is imported

**Comment by bradyhunsaker** [comment]
Thanks. I managed to run google-java-format from a separate download rather than using gradlew. Of course it just made the change you gave.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ParentStationValidator.java (hunk 3 above)
Comment (bradyhunsaker) [reply main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ParentStationValidator.java]: I changed it to import INFO, as it already does with ERROR.

File: main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ParentStationValidator.java [hunk 4]
```diff
@@ -31,17 +34,20 @@
...
- * Validates `location_type` of the referenced `parent_station`.
+ * Validates parent stations are used and have the correct `location_type`.
  *
- * <p>Generated notice: {@link WrongParentLocationTypeNotice}.
+ * <p>Generated notices:
+ *
+ * <ul>
+ *   <li>{@link WrongParentLocationTypeNotice}.
+ *   <li>{@link UnusedParentStationNotice}.
  */
```
Comment (davidgamez) [inline main/src/main/java/org/mobilitydata/gtfsvalidator/validator/ParentStationValidator.java:44]: The `aggregateJavaDoc` it's failing due to a missing closing tag,
```suggestion
 * <ul>
 *   <li>{@link WrongParentLocationTypeNotice}.
 *   <li>{@link UnusedParentStationNotice}.
 * </ul>
 */
```

**Comment by bradyhunsaker** [comment]
Thanks for the documentation fix.

The end-to-end test is now failing in validate_gradle_wrapper, but the error message looks like it's something with the system itself rather than the PR. I'm not familiar with this testing setup, so I could be wrong about that.

**Review by davidgamez — APPROVED** [review]

**Comment by davidgamez** [comment]
> Thanks for the documentation fix.
> 
> The end-to-end test is now failing in validate_gradle_wrapper, but the error message looks like it's something with the system itself rather than the PR. I'm not familiar with this testing setup, so I could be wrong about that.

This is an intermittent issue downloading the gradle binary, I retried it and passed without issues. I'm approving and merging right now. Thanks for taking it to the end of the line!!

**Pull request merged**